src/H5private.h

```diff
@@ -1517,12 +1518,12 @@ H5_DLL herr_t H5CX_pop(bool update_dxpl_props);
 #define FUNC_LEAVE_API(ret_value)                                                                            \
     ;                                                                                                        \
     } /*end scope from end of FUNC_ENTER*/                                                                   \
-    if (api_ctx_pushed) {                                                                                    \
+    if (H5_LIKELY(api_ctx_pushed)) {                                                                         \
```

Comment:
This will slow down FUNC_ENTER_API_NOINIT. Do we have a sense for how much? This applies to all passthrough VOL calls. Any other use cases we know of that use these NOINIT API functions heavily?
Functions (besides H5VL): H5dont_atexit, H5check_version, H5close, H5allocate_memory, H5resize_memory, H5free_memory, H5is_library_threadsafe, H5is_library_terminating, H5FDperform_init

Reply:
I guess your link suggests the simple benchmark runs slightly faster than half the speed of the default case. This is almost certainly (very slightly) good for everyone but passthrough VOLs, though I'm inclined to take it out.

Reply:
Good points, I'll revert switching to H5_LIKELY(api_ctx_pushed) in FUNC_LEAVE_API

Reply:
> Good points, I'll revert switching to H5_LIKELY(api_ctx_pushed) in FUNC_LEAVE_API

Actually, I looked at this again and don't understand how this change affects FUNC_ENTER_API_NOINIT, which uses FUNC_LEAVE_API_NOINIT, not this macro.